source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -253,16 +255,16 @@ std::string Utility::getSubjectFromCertificate(X509& cert) {
   return getRFC2253NameFromCertificate(cert, CertName::Subject);
 }
 
-int32_t Utility::getDaysUntilExpiration(const X509* cert, TimeSource& time_source) {
+absl::optional<size_t> Utility::getDaysUntilExpiration(const X509* cert, TimeSource& time_source) {
   if (cert == nullptr) {
-    return std::numeric_limits<int>::max();
+    return absl::make_optional(std::numeric_limits<int>::max());
   }
   int days, seconds;
   if (ASN1_TIME_diff(&days, &seconds, currentASN1_Time(time_source).get(),
                      X509_get0_notAfter(cert))) {
-    return days;
+    return days >= 0 ? absl::make_optional(days) : absl::nullopt;
```

Comment:
Just after the cert expires, won't days be 0 with seconds negative? I think we need to check both to determine if the cert is expired.

Reply:
Sure.